File: codeforces/41A/cpp/41a.cpp
#include <iostream>
#include <string>
using namespace std;

bool correct(const string &s, const string &t)
{
    if (s.length() != t.length())
        return false;

    int len = s.length();
    for (int i = 0; i < len; i++) {
        if (s[i] != t[len-i-1]) {
            return false;
        }
    }

    return true;
}

int main()
{
    string s, t;
    cin >> s; 
    cin >> t;
    
    if (correct(s, t)) {
        cout << "YES" << endl;
    }
    else {
        cout << "NO" << endl;
    }
}

File: codeforces/41A/go/41a.go
package main

import (
    "fmt"
    )

func correct(s, t string) bool {
    lens := len(s)
    lent := len(t)
    if lens != lent {
        return false
    }

    for i := 0; i < lens; i++ {
        if s[i] != t[lens-i-1] {
            return false
        }
    }

    return true
}

func main() {
    var s, t string
    fmt.Scan(&s)
    fmt.Scan(&t)

    if correct(s, t) {
        fmt.Println("YES")
    } else {
        fmt.Println("NO")
    }
}
